### PR TITLE
:sparkles: Update _validate_documents to log file patterns and add error handling

### PR DESCRIPTION
**Pull request by aakankshaduggal**
Addresses #2306 

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the
  [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary).
- [ ] [Changelog](https://github.com/instructlab/instructlab/blob/main/CHANGELOG.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Functional tests have been added, if necessary.
- [ ] E2E Workflow tests have been added, if necessary.

### CODE REVIEW

**Review by khaledsulayman — CHANGES_REQUESTED** [review]
Could you add a test under `tests/test_lab_diff.py`?

File: src/instructlab/utils.py
Comment (jaideepr97) [inline src/instructlab/utils.py:302]: PDFs have a magic number, so we can determine if the file is really a pdf by checking the first 4 bytes of the header 
should we use that rather than rely on the file extension here?

Comment (jaideepr97) [inline src/instructlab/utils.py:302]: to be more future facing it may be better to define a slice of accepted formats and loop through it rather than add new if-else conditions for each new file format IMO 
that way extending to new formats will be easier and we don't repeat code

**Review by jaideepr97 — COMMENTED** [review]
thanks for the PR @aakankshaduggal!

File: src/instructlab/utils.py
Comment (nathan-weinberg) [reply src/instructlab/utils.py:302]: Personally I'm fine with the file extension, but no issue either way

Comment (nathan-weinberg) [reply src/instructlab/utils.py:302]: Agreed on this point

**Review by nathan-weinberg — DISMISSED** [review]
+1 to @khaledsulayman on adding a test and @jaideepr97 for changing the compatible format check to a loop rather than a series of if-else statements

File: src/instructlab/utils.py
Comment (cdoern) [inline src/instructlab/utils.py:295]: should this be info?

File: src/instructlab/utils.py
Comment (cdoern) [inline src/instructlab/utils.py:311]: `click.secho` is usually what we use for errors with `fg=red`

**Review by cdoern — COMMENTED** [review]
just some general comments. looks good!

**Comment by nathan-weinberg** [comment]
Putting on `hold` until commits are squashed

File: src/instructlab/utils.py
Comment (cdoern) [inline src/instructlab/utils.py:305]: ok so, you might need to assert the type of this before using it. Either that, or assert the type of the `file_types` dict above before instantiating it. 

you can assert the type of `file_info` here by doing `assert` or `isinstance`

Comment (cdoern) [reply src/instructlab/utils.py:305]: this is the only issue I think

**Review by cdoern — COMMENTED** [review]
I think this is the main issue causing mypy to fail

File: src/instructlab/utils.py
Comment (cdoern) [reply src/instructlab/utils.py:321]: decided to keep debug

File: src/instructlab/utils.py
Comment (cdoern) [reply src/instructlab/utils.py:324]: fixed

File: src/instructlab/utils.py
Comment (nathan-weinberg) [inline src/instructlab/utils.py:260]: I don't think this should be called `open_file` - it's a misleading function name, the purpose does not seem to be actually opening a file but validating whether or not a knowledge document is valid.

**Review by nathan-weinberg — COMMENTED** [review]
`temp_repo` directory should also be called something else - I assume based on the code it's supposed to be a test knowledge repo, so just call it that - `test_knowledge_repo`

Vague naming will lead to confusion down the road.

**Comment by aakankshaduggal** [comment]
> `temp_repo` directory should also be called something else - I assume based on the code it's supposed to be a test knowledge repo, so just call it that - `test_knowledge_repo`
> 
> Vague naming will lead to confusion down the road.

@nathan-weinberg 
Since this is a temp taxonomy repo, should we just name it `temp_taxonomy_repo`

File: src/instructlab/utils.py
Comment (nathan-weinberg) [reply src/instructlab/utils.py:260]: Yes thank you!

**Comment by khaledsulayman** [comment]
@merifygio retest

**Comment by khaledsulayman** [comment]
@mergifyio retest

**Comment by khaledsulayman** [comment]
@mergifyio refresh

**Comment by aakankshaduggal** [comment]
@Mergifyio refresh

**Comment by cdoern** [comment]
I could be wrong, but I have never seen anyone use the mergify refresh cmd. If you have made changes to your branch or specific tests are failing just push a commit up. The test failures here are really strange

**Comment by cdoern** [comment]
I re-triggered the tests and they seem to be running

**Review by cdoern — APPROVED** [review]
lgtm, thanks for working with us to get this passing CI :)

**Review by khaledsulayman — APPROVED** [review]

**Review by jaideepr97 — APPROVED** [review]

**Comment by cdoern** [comment]
@Mergifyio rebase

**Comment by cdoern** [comment]
@Mergifyio rebase

**Comment by cdoern** [comment]
we shouldn't merge until https://github.com/instructlab/instructlab/actions/runs/11109226470/job/30864229273 passes. Once it does I will merge manually since CI is broken

**Comment by cdoern** [comment]
e2e passed, and this is rebased ontop of main. Merging since mergify is broken

**Pull request merged**
